dev/com.ibm.ws.kernel.feature.core/src/com/ibm/ws/kernel/feature/internal/FeatureManager.java

```diff
@@ -1503,9 +1507,11 @@ protected boolean updateFeatures(WsLocationAdmin locService,
         boolean appForceRestartSet = false;
         final boolean sameJavaSpecVersion = sameJavaSpecVersion();
         try {
-            if (areConfiguredFeaturesGood(newConfiguredFeatures, newConfiguredPlatforms) && sameJavaSpecVersion) {
+            if (featureRepository.areConfiguredFeaturesGood(newConfiguredFeatures, newConfiguredPlatforms, platformEnvironmentVariable) &&
```

Comment:
should any of this be under beta flag?

Reply:
Hi; the 'areConfiguredFeaturesGood' method has a beta guard on the changed code.
I moved the method to the repository since the checks are all repo/ feature based.  It's the same, plus extra checks against the platform when beta is enabled.